### PR TITLE
New Mercury Models

### PR DESCRIPTION
**Pull request by Capkirk123**
Replace the FASA Mercury with BDB Mercury. Currently the parachute model is bugged, this will have to wait until it is fixed in BDB to release.

### CODE REVIEW

**Comment by lpgagnon** [comment]
I don't see a rp-0 counterpart of this PR; are you planning to make one?

(I notice this is adding RP0conf = true to the new parts; probably shouldn't)

**Comment by lpgagnon** [comment]
My notes from a couple of reentry tests:
- staging the LES decouples but doesn't activate the motor
- 'landing and control module' PAW has a "deploy goo experiment" button; it opens a little trap door that reveals what indeed looks like a small goo container. Not sure what to do with this in RO...
- 'landing and control module' PAW has two mysterious 'Status: locked' lines
- 'landing and control module' has a single staging action that activates chute, decouples the top node, AND activates RCS; decoupling at the same time as enabling rcs doesn't make much sense, and decoupling at same time as parachute means no staging the chute early to arm it. Since a part is limited to 1 staging action (I think), there may not be a better solution to this than 'don't use staging events, activate the different modules manually'.
- capsule and 'landing and control module' both run on hydrazine but have a magical 267s Isp
- no IVA?
- the destabilizing flap has a much lower skinMaxTemp than other parts, but a much _higher_ Internal max; seems odd. swapped values maybe?
- the destabilizing flap defaults to a Deploy Angle of 160, but the slider actually has a max of 100
- with the latest, harsher-heating FAR, heating behavior looks essentially identical to current ROC Mercury. A 200km Ap, -400km Pe reentry is barely survivable, bringing internal temp to 730K (not an issue, this is maintaining status quo)
- drogue chute doesn't autocut? (I was waiting for that as a cue to stage the main chute/decoupler)
- (main chute is upside down; already known)

**Comment by lpgagnon** [comment]
And some extra notes from the VAB:
- capsule+controlmodule+noseunit is 232kg lighter than current ROC Mercury capsule+nosefairing+chute. 869kg vs 637kg with empty tank and ablator removed
- new capsule has 66L in its SM tank, old capsule has 128L

**Comment by Capkirk123** [comment]
- The LES has a separate abort and separation motor, in my testing it seemed o work fine?

- I'm not sure how to get rid of the animation for the goo capsule. I'll probably just name it something else if I can't find a solution

- I'm not sure what to do about the landing and control module. The parachute is meant to be a separate part, but RealChutes refuses to deploy the chute if its a separate part because it doesn't pass the occlusion check. I've been trying to get info on how to make realchute ignore the occlusion check from starwaster, but he hasn't answered.

- The drogue chute doesn't autocut because its meant to pull away the nosecone. I suppose thats not very clear though

- Heat Shields are undergoing a revamp

- The difference in volume is because Kerbalism/TAC adds extra volume for life support. This capsule does not currently have a Kerbalism config, so it is missing the extra volume.

**Comment by lpgagnon** [comment]
The LES description _says_ staging should both decouple and activate the engine; it doesn't, it only decouples. I haven't tested manual activation; that's a different use case (abort vs normal autostaging)

volume: that makes sense

**Comment by Capkirk123** [comment]
LES, RCS and mass issues should be fixed now

**Comment by lpgagnon** [comment]
LES: yes, fixed; staging activates and decouples. New bug: the plume is permanently active, even before staging.

RCS: Isp looks saner now, and using HTP matches current ROC-Mercury

mass: is now higher, but still low compared to current ROC-Mercury. Current ROC-Mercury, capsule+chute+nosecone 869kg; ROC-MercuryBDB, capsule+rcsblock+nose 709kg (with empty tanks & no ablator in both cases). The smaller tank might account for some of the difference, but probably not all.

One weirdness: the 'dry mass' shown in the pod PAW is _higher_ than the actual minimum part mass: 685kg vs 589kg. Whereas ROC-Mercury 'dry mass' is slightly lower than actual minimum mass, 860 vs 869

**Comment by Capkirk123** [comment]
Despite my best attempts, the parachute canopy is still inverted. However, all other issues have been fixed an masses should be consistent within a few kilograms between the new and old Mercury parts. If I can find a fix for the main chute, I'll make the other PRs for this capsule.

**Comment by Capkirk123** [comment]
The heatshield has been separated from the capsule. Since Mercury doesn't have a lifting reentry its pretty safe even without the occlusion fixes. Parachutes have also been fixed because I'm a massive idiot who was editing the wrong part.

**Comment by lpgagnon** [comment]
My notes with the latest changes:
- the retro motors' thrust curve makes MJ think it's going to burn for 2000s, but works fine in actual use
- the "destabilizing flap" still defaults at 160 Deploy Angle (even though the max is 100); harmless?
- the retro tub decoupler's Staging is disabled by default, seems odd
- reentry temperatures are reasonable; 27 ablator left seems like reasonable margin
- parachute works!

Looks good to me.

File: GameData/ROCapsules/PartConfigs/MercuryBDB/MercuryHS.cfg [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		allowAnimationWhileShielded = False
+		restrictedNode = bottom
+		disableAfterPlaying = true
+	}
+
+	//use new heat shield tagging system to configure the shield instead
+	resetHeatShieldMass = True
+	resetHeatShieldAblator = True
+	heatShieldTag = LEO
+	heatShieldDiameter = 1.89
```
Comment (DRVeyl) [inline GameData/ROCapsules/PartConfigs/MercuryBDB/MercuryHS.cfg:70]: Note that using these tags means RO will reset the mass and the maxTemp.  So declaring those fields in this PART config is not strictly necessary and may be misleading.  Not an issue for merging this, we've only just gone through the round of standardizing what engine configs look like, we haven't made it to the part definitions there, let alone for heatshields.

File: GameData/ROCapsules/PartConfigs/MercuryBDB/MercuryLES.cfg [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	// heatConductivity = 0.04 // 1/3 default
+	dragModelType = default
+	maximum_drag = 0.3
+	minimum_drag = 0.2
+	angularDrag = 2
+	crashTolerance = 15
+	breakingTorque = 30
+	breakingForce = 30
+	maxTemp = 673.15
+	skinMaxTemp = 773.15
```
Comment (DRVeyl) [inline GameData/ROCapsules/PartConfigs/MercuryBDB/MercuryLES.cfg:39]: @lpgagnon You should test this with the FAR + my heating updates.  Those might be low skin temps for something that will need to survive some convection during launch.  I'd definitely suggest leaving the heatConductivity = 0.04 commented out.

File: GameData/ROCapsules/PartConfigs/MercuryBDB/MercuryCM.cfg [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+			maxAmount = 5.7305
+		}
+
+		TANK
+		{
+			name = CarbonDioxide
+			amount = 0
+			maxAmount = 380
+		}
+	}
```
Comment (DRVeyl) [inline GameData/ROCapsules/PartConfigs/MercuryBDB/MercuryCM.cfg:478]: We *really* need a generic LifeSupport pass, that can work like the heatshield pass:  Set a flag, request a duration, request number of personnel, let the tool add these to the tank and create the converter.

**Comment by DRVeyl** [comment]
> The difference in volume is because Kerbalism/TAC adds extra volume for life support. This capsule does not currently have a Kerbalism config, so it is missing the extra volume.

Suggestion: Put the appropriate volume in the tank, and if TAC/Kerbalism configs want to consume some of it, they can, otherwise the player can fill it with whatever they deem appropriate.  Rather than having a different tank size with/without Kerbalism, TACLS.

**Pull request merged**